app/models/changeset/pull_request.rb

```diff
@@ -161,7 +161,21 @@ def message
 
   def section_content(section_title, text)
     # ### Risks or Risks followed by === / ---
-    desired_header_regexp = "^(?:\\s*#+\\s*#{section_title}.*|\\s*#{section_title}.*\\n\\s*(?:-{2,}|={2,}))\\n"
+    emoji_regex =
+      '(?:'\
+        '\p{RI}\p{RI}'\
+        '|\p{Emoji}'\
+          '(?:\p{EmojiModifier}'\
+          '|\uFE0F\u20E3?'\
+          '|[\uE0020-\uE007E]+\uE007F)?'\
+          '(?:\u200D\p{Emoji}'\
+            '(?:\p{EmojiModifier}'\
+            '|\uFE0F\u20E3?'\
+            '|[\uE0020-\uE007E]+\uE007F)?'\
+          ')*'\
+      ')' # translated from unicode standard: https://unicode.org/reports/tr51/#EBNF_and_Regex
+    desired_header_regexp = "^(?:\\s*#+\\s*#{emoji_regex}*\\s*#{section_title}.*"\
+      "|\\s*#{emoji_regex}*\\s*#{section_title}.*\\n\\s*(?:-{2,}|={2,}))\\n"
```

Comment:
this is a bit complex ... can we simplify by removing all the non-word chars left of the heading and just keep the old code ?
```
text.gsub!(/(^#+\s*)\W+/, "\\1")
```

Reply:
This bit of code fails this test case: https://github.com/zendesk/samson/pull/3944/files#diff-a1fa917b3ccda77ed24658b564d3e06572c84cf95d080491faa7f39a839aaba2R477-R484

Let me think of an alternative solution 🤔

Reply:
What about updating 
`desired_header_regexp = "^(?:\\s*#+\\s*#{section_title}.*|\\s*#{section_title}.*\\n\\s*(?:-{2,}|={2,}))\\n"`
to
`desired_header_regexp = "^(?:\\s*#+\\s*\\W*\\s*#{section_title}.*|\\s*\\W*\\s*#{section_title}.*\\n\\s*(?:-{2,}|={2,}))\\n"`